repl/src/main/scala/ammonite/repl/interp/Interpreter.scala

```diff
@@ -135,12 +136,26 @@ class Interpreter(prompt0: Ref[String],
     s"Syntax Error: $msg\n$locationString"
   }
 
-  def processCorrectScript(parsedCode: Parsed.Success[Seq[Seq[String]]],
+  def processCorrectScript(rawParsedCode: Parsed.Success[Seq[( String, Seq[String])]],
                            evaluate: EvaluateCallback): Seq[ImportData] = {
 
+    var offset = 0
+    var parsedCode: Seq[( String, Seq[String])] = Seq()
+    for( (comment, code) <- rawParsedCode.get.value ){
+      val ncomment = comment + "\n"*offset
+      offset = offset + comment.count(_ == '\n') + code.foldLeft(0){ (z,x) =>
+        z + x.count(_ == '\n')
+      } + 1
+
+      parsedCode = parsedCode :+ (ncomment, code)
+    }
 
-    val blocks0 = Parsers.splitScript(hardcodedPredef).get.value ++ parsedCode.get.value
-    val blocks = blocks0.map(preprocess(_, ""))
+    val parsedHardcodedPredef  = Parsers.splitScript(hardcodedPredef).get.value
+    val blocks0 = parsedHardcodedPredef ++ parsedCode
+
+    val blocks = for{
+      (comment, code) <- blocks0
+    } yield preprocess(code,"",comment)
```

Comment:
The comment or empty lines at top of any code is caught by WL in compilationUnit parser, so splitScript return (String, Seq[String]) that is comments separately, since preprocessor recombines the parsed values into code it skips comments that is why I keep comments separately and combine them in complete method